tests/unit/container/test_with_protocols.py

```diff
@@ -0,0 +1,38 @@
+from typing import Protocol
+
+import pytest
+
+from dishka import Provider, Scope, WithProtocols, make_container
+from dishka.exceptions import NoFactoryError
+
+
+class AProtocol(Protocol):
+    pass
+
+
+class BProtocol(Protocol):
+    pass
+
+
+class C(AProtocol, BProtocol):
+    pass
+
+
+def test_get_parents_protocols() -> None:
+    provider = Provider(scope=Scope.APP)
+    provider.provide(C, provides=WithProtocols[C])
+    container = make_container(provider)
+
+    assert (
+        container.get(BProtocol)
+        is container.get(AProtocol)
+    )
+
+
+def test_get_by_not_protocol() -> None:
+    provider = Provider(scope=Scope.APP)
+    provider.provide(C, provides=WithProtocols[C])
```

Comment:
We need either another naming or another logic.
For `WithParents`  we provide class itself together with parents. The same is expected here if we keep the prefix `With`